### ISSUE TITLE
Windows multi-stage builds broken in 19.03.1

### ISSUE THREAD

**Issue by pbering** [docker/for-win#4349]
- [x] I have tried with the latest version of my channel (Stable or Edge)
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: `FBE0859A-7747-4099-9C4D-E97F0513F17F/20190731191818`

### Expected behavior

Able to build multi-stage Windows images

### Actual behavior

Fails with:

```
Step 6/7 : COPY --from=builder ["C:/install/test.txt", "C:/install/"]
failed to copy files: mkdir \\?: The filename, directory name, or volume label syntax is incorrect.
```

### Information

- Multi-stage builds was working fine on previous stable version: 18.09.0
- Windows Version: Windows 10 Enterprise 1903, build 18362.239
- Docker Desktop Version: 2.1.0.0

### Steps to reproduce the behavior

Can be reproduced with this dockerfile:

```Dockerfile
FROM mcr.microsoft.com/windows/servercore:1903 as builder

RUN MKDIR C:\install
RUN ECHO 'test' > C:\install\test.txt
RUN DIR C:\install

FROM mcr.microsoft.com/windows/servercore:1903

# FAILS!
COPY --from=builder ["C:/install/test.txt", "C:/install/"]

RUN DIR C:\install
```

Build with: `docker image build --tag test .`

**Comment by pbering** [comment]
Possible workaround is to ensure that the destination path exist **before** COPY: `RUN MKDIR C:\install`

**Comment by StefanScherer** [comment]
This seems a change in the Windows Docker engine, I can reproduce it on a Windows Server 2019. The docker build works with Docker Engine Enterprise 18.09.8, but fails with 19.03.1

**Comment by thaJeztah** [comment]
ping @tonistiigi @tiborvass 

/cc @ddebroy

**Comment by thaJeztah** [comment]
Actually looks like the behavior is correct if the target directory is missing; https://docs.docker.com/engine/reference/commandline/cp/

![3E907F53-7252-4245-83BF-C8D2402A6E8C](https://user-images.githubusercontent.com/1804568/62327414-d3131e00-b4b0-11e9-8161-60695b617929.jpeg)


The error message is confusing though

**Comment by mikeparker** [comment]
So this is a breaking change and should be listed in docs for docker desktop I guess?

**Comment by pbering** [comment]
hmm I don't agree that this is behavior is correct... @thaJeztah why are you referencing the CP documentation? The COPY documentation https://docs.docker.com/engine/reference/builder/#copy clearly states `If <dest> doesn’t exist, it is created along with all missing directories in its path.`

**Comment by thaJeztah** [comment]
Hm.. you're right; looks like there's different semantics (or at least documented) for `COPY` than `docker cp`

**Comment by thaJeztah** [comment]
@tonistiigi @tiborvass could you have a look at this one?

**Comment by pbering** [comment]
Found another quirk related to this: `COPY --from=builder ["C:/install", "C:/install"]` actual works, so it seems that this issue only happens when the SRC is one or more files.

**Comment by pbering** [comment]
Fixed in 19.03.2 and verified that it now works again, thank you!

**Comment by thaJeztah** [comment]
Thanks for confirming it's fixed 👍
